meilisearch-lib/src/options.rs

```diff
@@ -24,25 +24,69 @@ pub struct IndexerOpts {
     /// In case the engine is unable to retrieve the available memory the engine will
     /// try to use the memory it needs but without real limit, this can lead to
     /// Out-Of-Memory issues and it is recommended to specify the amount of memory to use.
-    #[structopt(long, default_value)]
+    #[clap(long, default_value_t)]
     pub max_memory: MaxMemory,
 
     /// The name of the compression algorithm to use when compressing intermediate
     /// Grenad chunks while indexing documents.
     ///
     /// Choosing a fast algorithm will make the indexing faster but may consume more memory.
-    #[structopt(long, default_value = "snappy", possible_values = &["snappy", "zlib", "lz4", "lz4hc", "zstd"])]
+    #[clap(long, default_value = "snappy", possible_values = &["snappy", "zlib", "lz4", "lz4hc", "zstd"])]
     pub chunk_compression_type: CompressionType,
 
     /// The level of compression of the chosen algorithm.
-    #[structopt(long, requires = "chunk-compression-type")]
+    #[clap(long, requires = "chunk-compression-type")]
     pub chunk_compression_level: Option<u32>,
 
     /// Number of parallel jobs for indexing, defaults to # of CPUs.
-    #[structopt(long)]
+    #[clap(long)]
     pub indexing_jobs: Option<usize>,
 }
 
+#[derive(Debug, Clone, Parser, Default)]
+pub struct SchedulerConfig {
+    /// enable the autobatching experimental feature
+    #[clap(long, hide = true)]
+    pub enable_autobatching: bool,
+    // The maximum number of updates that can be batched together. If None, this is unlimited. A
+    // value of 0 is interpreted as 1.
+    #[clap(long, requires = "enable-autobatching", hide = true)]
+    pub max_batch_size: Option<usize>,
+    // The maximum number of documents in a document batch. Since batch must contain at least one
+    // batch for the schedulet to be able to make progress, this is ignored if a single update contains
+    // more updates that what is specified.
```

Comment:
```suggestion
    // The maximum number of documents in a document batch. Since the scheduler needs
    // at least one update to make progress, the number of documents in a batch will always
    // be at least the size of the first update to batch.
```